[ao] disabling dynamic conv/convT ops

HDCharles opened the pull request:
Summary: as mentioned in https://github.com/pytorch/pytorch/issues/70480 the dynamic conv ops are currently missing a key feature to bring their performance in line with other dynamic ops, this diff disables conv/convT from being automatically quantized with convert dynamic

Test Plan: buck test //caffe2/test:quantization --test-selectors test_quantized_module#TestDynamicQuantizedModule

Differential Revision: D33511152

vkuzo commented on torch/ao/quantization/quantization_mappings.py:
```diff
@@ -112,12 +112,13 @@
-    nn.ConvTranspose1d: nnqd.ConvTranspose1d,
-    nn.ConvTranspose2d: nnqd.ConvTranspose2d,
-    nn.ConvTranspose3d: nnqd.ConvTranspose3d,
+    # Don't want to enable these by default because the performance is poor compared to other dynamic ops
```
nit: do we mean "accuracy"?  performance is usually used for "inference time" in this context

HDCharles replied:
yeah, i'll fix that

vkuzo commented:
should we also add a warning to the constructors of the Python modules?  We can remove it once the more accurate kernels are there.

HDCharles committed:
[ao] disabling dynamic conv/convT ops (#71110)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/71110

as mentioned in https://github.com/pytorch/pytorch/issues/70480 the dynamic conv ops are currently missing a key feature to bring their performance in line with other dynamic ops, this diff disables conv/convT from being automatically quantized with convert dynamic

Test Plan: buck test //caffe2/test:quantization --test-selectors test_quantized_module#TestDynamicQuantizedModule

Reviewed By: vkuzo

Differential Revision: D33511152

fbshipit-source-id: 6f4d4ec2684426b524d1af722c836561129500ae
